qiskit/visualization/tools/pi_check.py

```diff
@@ -124,5 +131,9 @@ def pi_check(inpt, eps=1e-6, output='text', ndigits=5):
 
         return str_out
     # nothing found
-    str_out = '%.{}g'.format(ndigits) % inpt
+    if complex_check == 'True':
```

Comment:
The `isinstance(inpt, complex)` from line 53 could be moved here and that would avoid this `complex_check` flag, right?

Reply:
This alone would have not worked because the code "inpt = float(inpt)" appears way above on the top and any complex number as inpt gets errored out at this stage.